modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -479,6 +479,38 @@ public void deployObjectDefinition(ObjectDefinition objectDefinition) {
 		}
 	}
 
+	@Override
+	public ObjectDefinition enableAccountEntryRestricted(
+			ObjectRelationship objectRelationship)
+		throws PortalException {
+
+		ObjectDefinition objectDefinition1 = getObjectDefinition(
+			objectRelationship.getObjectDefinitionId1());
+
+		if (!Objects.equals(objectDefinition1.getShortName(), "AccountEntry")) {
+			throw new ObjectDefinitionAccountEntryRestrictedException(
+				"It is only possible to restrict custom object definitions " +
+					"with account entry");
```

Comment:
Hey Brian,
We didn't add another validation to this because we realized that this case is already covered by Object Relationships, as it can only be created with up to 1 system object. So if the objectDefinition1 it's already a system object the objectDefinition2 will never be a system.